poc/mastic.py

```diff
@@ -305,15 +308,14 @@ def prep_init(
             onehot_check + counter_check + payload_check,
         ).next(PROOF_SIZE)
 
-        # Concatenate the output shares into one aggregatable output,
-        # applying the FLP truncation algorithm on each FLP measurement
-        # share.
-        truncated_out_share = []
-        for val_share in out_share:
-            truncated_out_share += [val_share[0]] + \
-                self.flp.truncate(val_share[1:])
+        flattened_out_share = []
+        for out_share in out_shares:
+            # The VIDPF transform truncates at every level but the first.
+            if level == 0:
+                out_share = self.truncate(out_share)
+            flattened_out_share += out_share
```

Comment:
Why do we need to truncate the first level? Why not just pass `out_share` here?

Reply:
I got this part, what I'm missing is the why do we need to truncate? We already evaluated right, so why truncate after? (It might be obvious but I didn't get it from the description)

Reply:
Oh for the path checks! Right?

Reply:
Are you asking why we need to truncate at all?

Reply:
I understand the truncation is to reduce the public share size - what I was missing is why we need to truncate the first level after we evaluate it.